src/data_structures/fmindex.rs

```diff
@@ -168,18 +183,46 @@ impl FMDIndex {
     ///   less memory usage, but worse performance)
     pub fn new(bwt: BWT, k: usize) -> Self {
         let mut alphabet = dna::n_alphabet();
-        alphabet.insert(b'$');
-        assert!(
-            alphabet.is_word(&bwt),
-            "Expecting BWT over the DNA alphabet (including N) with the sentinel $."
-        );
+        alphabet.insert(SENT_CHAR);
+        assert!(alphabet.is_word(&bwt),
+                "Expecting BWT over the DNA alphabet (including N) with the sentinel $.");
 
         FMDIndex {
             fmindex: FMIndex::new(bwt, k, &alphabet),
-            revcomp: dna::RevComp::new()
+            revcomp: dna::RevComp::new(),
         }
     }
 
+    /// Find longest non-encompassing prefix match
+    pub fn longest_prefix_match(&self, pattern: &[u8], min_overlap: usize) -> Option<BiInterval> {
+        let exact_match_interval = self.fmindex.backward_search(pattern.iter());
```

Comment:
So, one of the design principles of Rust-Bio is to use the state of the
art algorithm if it exists. So, since there are publications available,
I would like the implementation to follow them. If your implementation
is pretty similar already, it should be easy to adapt. I'm really sorry
for bothering you with this, but I just want to codebase to follow this
rule whenever possible. This is also important to give credit to the
right people in the docs, instead of slightly reinventing the wheel.
Also, note that both pubs are about the FMIndex, not the FMDIndex. I am
not sure about the implications of doing this on the FMDIndex, nor am I
sure if this is wanted in your case (it will also consider reverse
complements).
On Tue, 2016-03-01 at 09:31 -0800, Taylor C. wrote:
> In src/data_structures/fmindex.rs:
> >          }
> >      }
> >  
> > +    /// Find longest non-encompassing prefix match
> > +    pub fn longest_prefix_match(&self, pattern: &[u8],
> min_overlap: usize) -> Option {
> > +        let exact_match_interval =
> self.fmindex.backward_search(pattern.iter());
> What I'm doing is very similar to the algorithm given on page 33 of
> the Sanger paper.
> —
> Reply to this email directly or view it on GitHub.